src/features/hover.zig

```diff
@@ -18,6 +18,9 @@ pub fn hoverSymbol(server: *Server, decl_handle: Analyser.DeclWithHandle, markup
     const tracy_zone = tracy.trace(@src());
     defer tracy_zone.end();
 
+    var temp_arena = std.heap.ArenaAllocator.init(server.allocator);
```

Comment:
why create a new arena allocator here instead of using `server.arena`?

Reply:
memory allocated with `server.arena` is only guaranteed to live until you finished handling a request. If you need something that lives until the server is stopped, you would use `server.allocator` and manage its lifetime. `server.arena` is exactly what you need.